tls/s2n_server_extensions.c

```diff
@@ -45,6 +45,11 @@
 
 int s2n_server_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *out)
 {
+
+    if (conn->secure.cipher_suite->key_exchange_alg == NULL) {
+        return 0;
+    }
```

Comment:
hmm, so you don't want to send any extensions when this is NULL?  that seems like overkill.

perhaps you simply mean to change line 100:

```
if (conn->secure.cipher_suite->key_exchange_alg) {
  GUARD(s2n_kex_write_server_extension(conn->secure.cipher_suite->key_exchange_alg, conn, out));
}
```

Reply:
Thanks Matthew for reviewing :) I think we also need to change below line:
```
    if (conn->secure.cipher_suite->key_exchange_alg) {
        total_size += s2n_kex_server_extension_size(conn->secure.cipher_suite->key_exchange_alg, conn);
    }
```

Reply:
that too - good spot ;)